lib/tasks/database_updates/reset_permalocked_users.rake

```diff
@@ -0,0 +1,18 @@
+namespace :database_updates do
+  desc 'Reset Permalocked users'
+  task :reset_permalocked_users => :environment do
+    publisher_ids = []
+    PublisherStatusUpdate.select(:status, :publisher_id).where(status: "locked").group(:publisher_id, :status).having("count(*) > 1").each do |psu|
+      publisher_ids.append(psu.publisher_id)
+    end
+
+    publisher_ids -= TwoFactorAuthenticationRemoval.pluck(:publisher_id)
+
+    Publisher.includes(:status_updates).where(id: publisher_ids).find_each do |publisher|
+      if publisher.locked?
+        previous_status = publisher.status_updates.select { |status_update| status_update.status != PublisherStatusUpdate::LOCKED }.first.status
```

Comment:
Are these ordered properly so that we want the first one? If we want the latest non-locked status, would we reverse the order and then choose first?

Reply:
Yes, it's sorted by default: https://github.com/brave-intl/publishers/blob/d5b83e4e7c015854ff2568ab4401c9b6f2c458cc/app/models/publisher.rb#L42